packages/main/src/plugin/provider-registry.spec.ts

```diff
@@ -1145,6 +1149,86 @@ test('should send events when stopping a container connection', async () => {
   expect(onAfterDidUpdateContainerConnectionCalled).toBeTruthy();
 });
 
+test('should send events when container connection status change', async () => {
+  const provider = providerRegistry.createProvider('id', 'name', {
+    id: 'internal',
+    name: 'internal',
+    status: 'installed',
+  });
+  const connection: ProviderContainerConnectionInfo = {
+    name: 'connection',
+    displayName: 'connection',
+    type: 'docker',
+    endpoint: {
+      socketPath: '/endpoint1.sock',
+    },
+    status: 'stopped',
+    vmType: {
+      id: 'libkrun',
+      name: 'libkrun',
+    },
+  };
+
+  const startMock = vi.fn();
+  const stopMock = vi.fn();
+  const containerProviderConnection: ContainerProviderConnection = {
+    name: 'connection',
+    displayName: 'connection',
+    type: 'docker',
+    lifecycle: {
+      start: startMock,
+      stop: stopMock,
+    },
+    endpoint: {
+      socketPath: '/endpoint1.sock',
+    },
+    status() {
+      return 'started';
+    },
+    vmType: 'libkrun',
+  };
+  provider.registerContainerProviderConnection(containerProviderConnection);
+
+  let onBeforeDidUpdateContainerConnectionCalled = false;
+  providerRegistry.onBeforeDidUpdateContainerConnection(event => {
+    expect(event.connection.name).toBe(connection.name);
+    expect(event.connection.type).toBe(connection.type);
+    expect(event.status).toBe('stopped');
+    onBeforeDidUpdateContainerConnectionCalled = true;
+  });
+  let onDidUpdateContainerConnectionCalled = false;
+  providerRegistry.onDidUpdateContainerConnection(event => {
+    expect(event.connection.name).toBe(connection.name);
+    expect(event.connection.type).toBe(connection.type);
+    expect(event.status).toBe('stopped');
+    onDidUpdateContainerConnectionCalled = true;
+  });
```

Comment:
Same as above, probably better to use a vitest mock for the event.

```ts
let onDidUpdateContainerConnectionCalledMock = vi.fn();
providerRegistry.onDidUpdateContainerConnection(onDidUpdateContainerConnectionCalledMock);

[...]

expect(onDidUpdateContainerConnectionCalledMock).toHaveBeenCalledOnce();
expect(onDidUpdateContainerConnectionCalledMock).toHaveBeenCalledWith({
  connection: expect.objectContaining({
    name: connection.name,
    type: connection.type,
  }),
  providerId: provider.id,
  status: 'stopped',
});
```

Checking with `expect` in a callback is error prone, as we may never call the callback, so we should probably provide a mock, and use expect on it, in the test itself, to ensure we are properly checking all  the conditions :+1: